R/helpers_knitr_GeneralModel.R

```diff
@@ -941,3 +962,201 @@ knit_print.LogisticLogNormalOrdinal <- function(
   }
   NextMethod(params = params)
 }
+
+# LogisticIndepBeta ----
+
+#' @description `r lifecycle::badge("experimental")`
+#' @rdname knit_print
+#' @export
+#' @method knit_print LogisticIndepBeta
+knit_print.LogisticIndepBeta <- function(
+    x,
+    ...,
+    use_values = TRUE,
+    fmt = "%5.2f",
+    params = NA,
+    tox_label = "DLAEs",
+    preamble = "The prior for &theta; is given by\\n",
+    asis = TRUE) {
+  assert_flag(asis)
+
+  y <- tidy(x)
+  z <- "e^{\\phi_1 + \\phi_2 \\cdot log(d)}"
+  posterior <- ModelParamsNormal(mean = c(x@phi1, x@phi2), cov = x@Pcov)
+  # knit_print.ModelParamsNormal expects no row or column names
+  rownames(posterior@cov) <- NULL
+  colnames(posterior@cov) <- NULL
+
+  rv <- paste0(
+    "A logistic log normal model will describe the relationship between dose and toxicity: ",
+    "$$ p(Tox | d) = f(X = 1 | \\theta, d) = \\frac{", z, "}{1 + ", z, "} $$\\n ",
+    "The prior is expressed in terms of pseudo data and, consequently, the number ",
+    " of cases and of ",
+    tox_label,
+    " need not be whole numbers.\n\nThe pseudo data are ",
+    "defined in the following table:\n\n",
+    paste0(
+      do.call(
+        function(x) {
+          kableExtra::kable_styling(
+            knitr::kable(x, col.names = c("Dose", "N", tox_label)),
+            bootstrap_options = c("striped", "hover", "condensed")
+          )
+        },
+        list(x = y$pseudoData)
+      ),
+      collapse = "\n"
+    ),
+    ifelse(
+      nrow(y$data) == 0,
+      "\n\nNo observed data has yet been recorded.\n",
+      paste(
+        "\n\nThe observed data are given in the following table:\n\n",
+        paste((do.call(knitr::kable, list(x = y$data))), collapse = "\n")
+      )
+    ),
+    knit_print(
+      posterior,
+      preamble = paste0(
+        "\n\nTogether, the pseudo and observed data give rise to ",
+        "the following posterior for the model parameters:\n\n"
+      ),
+      params = c("\\phi_1" = "phi1", "\\phi_2" = "phi2"),
+      theta = "\\phi",
+      asis = FALSE,
+      ...
+    ),
+    "\n\n"
+  )
+
```

Comment:
not sure why but the current snapshot is just an empty page

Reply:
There's something strange going on.  Test results are odd (see #822) and differ between GitHub and Apollo.  I am investigating.

Reply:
This appears to have been resolved.